src/CubeHandler.php

```diff
@@ -16,18 +17,55 @@
  */
 final class CubeHandler extends AbstractHttpClientHandler
 {
+    /**
+     * @var string|UriInterface
+     */
+    private $uri;
+
+    /**
+     * @param string|UriInterface $uri
+     * @param int|string $level The minimum logging level at which this handler will be triggered
+     */
     public function __construct(
         ClientInterface $client,
         RequestFactoryInterface $requestFactory,
+        $uri,
         $level = Logger::ERROR,
         bool $bubble = true
     ) {
         parent::__construct($client, $requestFactory, $level, $bubble);
+        $this->uri = $uri;
     }
 
-    // todo
-    public function createRequest(array $record): RequestInterface
+    protected function createRequest(array $record): RequestInterface
     {
-        return $this->requestFactory->createRequest('GET', 'what');
+        $date = $record['datetime'];
+
+        $data = ['time' => $date->format('Y-m-d\TH:i:s.uO')];
+        unset($record['datetime']);
```

Comment:
What is the purpose of the unset here and at line 49?

Reply:
Tbh I used the one from monolog as a reference...

Reply:
Fixed.